identity/src/keypair.rs

```diff
@@ -176,12 +178,12 @@ impl Keypair {
     /// Sign a message using the private key of this keypair, producing
     /// a signature that can be verified using the corresponding public key.
     #[allow(unused_variables)]
-    pub fn sign(&self, msg: &[u8]) -> Result<Vec<u8>, SigningError> {
+    pub fn sign(&self, msg: &[u8]) -> Result<Vec<u8>, Infallible> {
```

Comment:
Since we are returning `Ok` throughout the inner functions, should we go on and remove the `Result` here?

Reply:
We can. But what about upstream? there are a couple of functions that return a `Result` only because `sign` does so. Should we address them here? The biggest change will be noise security upgrade become infallible but the trait bound in swarm builder expect an associated `Error` type.

Reply:
We could change it so it is no longer expecting an associated error type

Reply:
Im not against making the change here. Thoughts on that @jxs ?

Reply:
yeah I think if it's api that doesn't make sense we should address it and take the next release cycle to release a new `libp2p-identity` release wdyt?